pydra/engine/submitter.py

```diff
@@ -30,7 +30,7 @@ def __call__(self, runnable, cache_locations=None):
         if cache_locations is not None:
             runnable.cache_locations = cache_locations
         if is_workflow(runnable):
-            runnable.submit_async(self)
+            self.submit_wf_async(runnable)
```

Comment:
we can extract the logic from `submit_wf_async` into here, since the only different condition is a workflow without state.

```python
if is_workflow(runnable) and runnable.state is None:
    self.loop.run_until_complete(runnable._run(self))
else:
    self.loop.run_until_complete(self.submit(runnable, return_task=True))
```